internal/pipe/webhook/webhook_test.go

```diff
@@ -0,0 +1,258 @@
+package webhook
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStringer(t *testing.T) {
+	require.Equal(t, Pipe{}.String(), "webhook")
+}
+
+func TestNoEndpoint(t *testing.T) {
+	ctx := context.New(config.Project{
+		Announce: config.Announce{
+			Webhook: config.Webhook{},
+		},
+	})
+	require.EqualError(t, Pipe{}.Announce(ctx), `announce: failed to announce to webhook: no endpoint url`)
+}
+
+func TestMalformedEndpoint(t *testing.T) {
+	ctx := context.New(config.Project{
+		Announce: config.Announce{
+			Webhook: config.Webhook{
+				EndpointURL: "httxxx://example.com",
+			},
+		},
+	})
+	require.EqualError(t, Pipe{}.Announce(ctx), `announce: failed to announce to webhook: malformed endpoint url`)
+}
+
+func TestAnnounceInvalidMessageTemplate(t *testing.T) {
+	ctx := context.New(config.Project{
+		Announce: config.Announce{
+			Webhook: config.Webhook{
+				EndpointURL:     "https://example.com/webhook",
+				MessageTemplate: "{{ .Foo }",
+			},
+		},
+	})
+	require.EqualError(t, Pipe{}.Announce(ctx), `announce: failed to announce to webhook: template: tmpl:1: unexpected "}" in operand`)
+}
+
+type WebHookServerMockMessage struct {
+	Response string    `json:"response"`
+	UUID     uuid.UUID `json:"uuid"`
+}
+
+func TestAnnounceWebhook(t *testing.T) {
+	responseServer := WebHookServerMockMessage{
+		Response: "Thanks for the announcement!",
+		UUID:     uuid.New(),
+	}
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			panic(err)
+		}
+		require.Equal(t, "webhook-test", string(body))
+
+		w.WriteHeader(http.StatusCreated)
+		w.Header().Set("Content-Type", "application/json")
+		err = json.NewEncoder(w).Encode(responseServer)
+		if err != nil {
+			panic(err)
+		}
+	}))
+	defer srv.Close()
+
+	ctx := context.New(config.Project{
+		ProjectName: "webhook-test",
+		Announce: config.Announce{
+			Webhook: config.Webhook{
+				EndpointURL:     srv.URL,
+				MessageTemplate: "{{ .ProjectName }}",
+			},
+		},
+	})
+	require.NoError(t, Pipe{}.Announce(ctx))
+}
+
+func TestAnnounceTLSWebhook(t *testing.T) {
+	responseServer := WebHookServerMockMessage{
+		Response: "Thanks for the announcement!",
+		UUID:     uuid.New(),
+	}
+
+	srv := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
why not

```suggestion
		require.NoError(t, err)
```

?

Reply:
I was not sure if its part of the test or a testsupporting failure 😄

Reply:
it works! there are several of those in this file, I think we can replace them all :D

Reply:
done